kotlin-codegen/compiler/src/test/java/com/squareup/moshi/KotlinCompilerCall.kt

```diff
@@ -166,4 +176,20 @@ class KotlinCompilerCall(var scratchDir: File) {
     throw IllegalStateException("no kotlin-annotation-processing-embeddable jar on classpath:\n  " +
         "${classpathFiles().joinToString(separator = "\n  ")}}")
   }
-}
\ No newline at end of file
+
+  /**
+   * Base64 encodes a mapping of annotation processor args for kapt, borrowed from
+   * https://kotlinlang.org/docs/reference/kapt.html#apjavac-options-encoding
```

Comment:
for fun, i looked at ObjectOutputStream and tried to make this work with Okio.
```kotlin
val body = Buffer()
body.writeInt(options.size)
for ((key, value) in options.entries) {
  body.writeShort(Utf8.size(key).toInt())
  body.writeUtf8(key)
  body.writeShort(Utf8.size(value).toInt())
  body.writeUtf8(value)
}

val result = Buffer()
result.writeShort(STREAM_MAGIC.toInt())
result.writeShort(STREAM_VERSION.toInt())
val bodySize = body.size()
when {
  bodySize <= 0xFF -> {
    result.writeByte(TC_BLOCKDATA.toInt())
    result.writeByte(body.size().toInt())
  }
  bodySize <= 0x7FFFFFFF -> {
    result.writeByte(TC_BLOCKDATALONG.toInt())
    result.writeInt(bodySize.toInt())
  }
  else -> TODO("More blocks.")
}
result.writeAll(body)

val readByteString = result.readByteString()
return readByteString.base64()
```

so much hidden knowledge in ObjectOutputStream!